File: Data Structures & Algorithms/validate-parentheses/submission-14.py
class Solution:
    def isValid(self, s: str) -> bool:
        pMap = {
            ')' : '(',
            ']' : '[',
            '}' : '{'
        }

        stack = []

        for p in s:
            if p not in pMap:
                stack.append(p)
                continue

            if stack:
                if stack[-1] != pMap[p]:
                    return False
                else:
                    stack.pop()
            else:
                return False
                
        return len(stack) == 0

File: Data Structures & Algorithms/daily-temperatures/submission-14.ts
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures: number[]): number[] {
        const n = temperatures.length
        const stack: number[][] = []
        const res = new Array(n).fill(0)

        for (let i = 0; i < n; i++) {
            while (stack.length && stack[stack.length - 1][0] < temperatures[i]) {
                const [t, tIndex] = stack.pop()
                res[tIndex] = i - tIndex
            }

            stack.push([temperatures[i], i])
        }

        return res
    }
}
